### ISSUE TITLE
Avoid creation of ~/.k8slens/extensions/ directory

### ISSUE THREAD

**Issue by ogarcia** [lensapp/lens#2494]
**Describe the bug**
Now the app is creating the directory `~/.k8slens/extensions/` instead of use `${XDG_CACHE_HOME}/k8slens/`.

**To Reproduce**
1. Open app

**Expected behavior**
That the app follows the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)-

**Environment (please complete the following information):**
- Lens Version: 4.2.1
- OS: Linux
- Installation method: [AUR](https://aur.archlinux.org/packages/lens/)

**Additional context**
As side note in Linux `$XDG_CACHE_HOME` may not be declared, in this case must fallback to `~/.cache`.

For example in BASH:
```bash
K8LENS_DIR=${XDG_CACHE_HOME:-~/.cache}/k8lens
```

Or in Python:
```python
k8lens_dir = os.path.join(os.environ.get('XDG_CACHE_HOME', os.path.join(os.path.expanduser('~'), '.cache')), 'k8lens')
```

**Comment by Nokel81** [comment]
I don't think it should be under `~/.cache` anyway. Since you can actually symlink your own folders into that folder (as an alternative form of extension installation usually used by extension developers).

**Comment by ogarcia** [comment]
In any case it must follow the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
 
You can use `${XDG_DATA_HOME}` (defaults to `~/.local/share`) but IMHO an enabled extensión can be downloaded again (is a non-essential data file) so use `~/.cache` not is bad idea.

Update: Take note that in these directories you must create a k8lens subdirectory, this is `~/.local/share/k8lens` or `~/.cache/k8lens`

**Comment by Nokel81** [comment]
Maybe I don't know enough about XDG but from reading that spec it seems to be about KDE in particular and not the applications that run on top of it.

**Comment by ogarcia** [comment]
XDG is a standard. All Linux applications must follow it, not only KDE ones. It's designed to make consistent profiles, like in MacOS with the `~/Library/Application Support/` and `~/Library/Caches/`.

**Comment by nevalla** [comment]
We tried to follow how VS Code is doing this and at least on Mac it's using `~/.vscode/extensions`. I assume it's doing the same on other environments too.

Now when investigating the topic more I can see there's quite old issue on VS Code repository where they propose to respect XDG standard too (https://github.com/microsoft/vscode/issues/3884).

**Comment by obvionaoe** [comment]
Most Linux users and applications follow the XDG standard. It would be great if Lens could follow it as well

**Comment by erayaydin** [comment]
Sorry for bumping this issue.

Related code change: https://github.com/lensapp/lens/blob/dfcb7c33309621237245127b34d67b5134743517/src/renderer/components/dock/create-resource/user-templates.injectable.ts#L17

We will so happy if this issue fixed. It's also effect to all other UNIX-like platforms.
